designer_v2/lib/features/dashboard/studies_table.dart

```diff
@@ -14,60 +17,146 @@ class StudiesTable extends StatelessWidget {
         ),
         StandardTableColumn(
           label: tr.studies_list_header_participants_enrolled,
-          columnWidth: const FlexColumnWidth(0.5),
+          columnWidth: MaxColumnWidth(FixedColumnWidth(statsCellWidth), const IntrinsicColumnWidth()),
+          sortable: true,
         ),
         StandardTableColumn(
           label: tr.studies_list_header_participants_active,
-          columnWidth: const FlexColumnWidth(0.5),
+          columnWidth: MaxColumnWidth(FixedColumnWidth(statsCellWidth), const IntrinsicColumnWidth()),
+          sortable: true,
         ),
         StandardTableColumn(
           label: tr.studies_list_header_participants_completed,
-          columnWidth: const FlexColumnWidth(0.5),
+          columnWidth: MaxColumnWidth(FixedColumnWidth(statsCellWidth), const IntrinsicColumnWidth()),
+          sortable: true,
         ),
       ],
       onSelectItem: onSelect,
       trailingActionsAt: (item, _) => getActions(item),
       buildCellsAt: _buildRow,
+      sortColumnPredicates: _sortColumns,
+      pinnedPredicates: pinnedPredicates,
       emptyWidget: emptyWidget,
     );
   }
 
+  int Function(Study a, Study b) get pinnedPredicates {
+    return (Study a, Study b) {
+      if (pinnedStudies.contains(a.id)) {
+        return -1;
+      } else if (pinnedStudies.contains(b.id)) {
+        return 1;
+      }
+      return 0;
+    };
+  }
+
+  List<int Function(Study a, Study b)?> get _sortColumns {
+    final predicates = [
+      (Study a, Study b) => 0, // do not sort pin icon
+      (Study a, Study b) => a.title!.compareTo(b.title!),
+      (Study a, Study b) => a.status.index.compareTo(b.status.index),
+      (Study a, Study b) => a.participation.index.compareTo(b.participation.index),
+      (Study a, Study b) => a.createdAt!.compareTo(b.createdAt!),
+      (Study a, Study b) => a.participantCount.compareTo(b.participantCount),
+      (Study a, Study b) => a.activeSubjectCount.compareTo(b.activeSubjectCount),
+      (Study a, Study b) => a.endedCount.compareTo(b.endedCount),
+    ];
+    return predicates;
+  }
+
   List<Widget> _buildRow(BuildContext context, Study item, int rowIdx, Set<MaterialState> states) {
     final theme = Theme.of(context);
 
     TextStyle? mutedTextStyleIfZero(int value) {
       return (value > 0) ? null : ThemeConfig.bodyTextBackground(theme);
     }
 
+    Icon icon(IconData iconData) {
+      return Icon(
+        iconData,
+        color: Colors.grey,
+        size: 25,
+      );
+    }
+
+    Widget getRespectivePinIcon(Set<MaterialState> state) {
+      final bool contains = pinnedStudies.contains(item.id);
+      final bool hovers = state.contains(MaterialState.hovered);
+      switch (hovers) {
+        case true:
+          return contains ? icon(MdiIcons.pinOff) : icon(MdiIcons.pin);
+        case false:
+          return contains ? icon(MdiIcons.pin) : const SizedBox.shrink();
+        default:
+          return const SizedBox.shrink();
+      }
```

Comment:
Can the default case even happen? Why not

```suggestion
      if (hovers) {
        return contains ? icon(MdiIcons.pinOff) : icon(MdiIcons.pin);
      } else {
        return contains ? icon(MdiIcons.pin) : const SizedBox.shrink();
      }
```